src/App/Document.cpp

```diff
@@ -3945,6 +3946,9 @@ std::vector<DocumentObject *> Document::addObjects(const char* sType, const std:
     objects.resize(objectNames.size());
     std::generate(objects.begin(), objects.end(),
                   [&]{ return static_cast<App::DocumentObject*>(type.createInstance()); });
+    // the type instance could be a null pointer, it is enough to check the first element
+    if (!objects[0])

```

Comment:
There's no code in this function guaranteeing that there is anything in the objectNames vector: it seems like to be on the safe side that should be getting checked someplace, otherwise this call will give unpredictable results.